### ISSUE TITLE
keras.model.predict doesn't work when using multiprocessing

### ISSUE THREAD

**Issue by u1s1QS** [tensorflow/tensorflow#45426]
I am using scipy.optimize.differential_evolution to optimize my autoencoder weights. The individuals are 20 combination of weights.

`result = differential_evolution(obj_func, bounds, args=args, strategy='best1bin', maxiter=2, popsize=20, tol=1e-6, mutation=(0.5, 1), recombination=0.3, seed=None, callback=printCurrentIteration, disp=True, polish=False, init=pop, atol=0, updating='deferred', workers=5)`

In my obj_func there is a part to get output from keras.model.predict(), here is my naive code:

```
def obj_func(x, *args):
         final_output = predict(x, data_p)
         ……
```

```
def predict(para, data_p): 
    new_weight_e1 = para[0:352].reshape(22, 16)
    new_weight_b1 = para[352:368].reshape(16, )
    new_weight_e2 = para[368:496].reshape(16, 8)
    new_weight_b2 = para[496:504].reshape(8, )
    new_weight_e3 = para[504:520].reshape(8, 2)
    new_weight_b3 = para[520:522].reshape(2, )
    new_weight_d1 = para[522:538].reshape(2, 8)
    new_weight_b4 = para[538:546].reshape(8, )
    new_weight_d2 = para[546:674].reshape(8, 16)
    new_weight_b5 = para[674:690].reshape(16, )
    new_weight_d3 = para[690:1042].reshape(16, 22)
    new_weight_b6 = para[1042:1064].reshape(22, )
    autoencoder.get_layer('e1').set_weights([(new_weight_e1), (new_weight_b1)])
    autoencoder.get_layer('e2').set_weights([(new_weight_e2), (new_weight_b2)])
    autoencoder.get_layer('out').set_weights([(new_weight_e3), (new_weight_b3)])
    autoencoder.get_layer('d1').set_weights([(new_weight_d1), (new_weight_b4)])
    autoencoder.get_layer('d2').set_weights([(new_weight_d2), (new_weight_b5)])
    autoencoder.get_layer('d3').set_weights([(new_weight_d3), (new_weight_b6)])

    output_layer_model = Model(inputs=autoencoder.input, outputs=decoded)
    final_output = output_layer_model.predict(data_p)
    
    return final_output
```

In function `predict()` I reshape the individual(weights), build the model and predict. 
In differential_evolution people can easily implement multiprocessing by setting 'workers' to more than 1. 
When I set it to 1, there is no problem. But when I set it to e.g. 5, want to predict parallel, the progress stopped here:

` final_output = output_layer_model.predict(data_p)`

No errors, it was still running but did not go further (didn't get the `final_ouput`). (The code before this line goes parallel without problem)

What I've tried:
1.
add` os.environ["CUDA_VISIBLE_DEVICES"] = "-1"`, only use multiple CPUs to predict, didn't work for me.
2.
compile the model inside the function, also didn't work.
3.
save the weights as .h5 file and load, also didn't work.

Again, it works well when only using 1 CPU. So I think the problem might be `keras.model.predict()` and `multiprocessing` but I didn't figure it out. I've done some search but didn't find a solution, maybe I didn't grasp the point because my English is not good. So any suggestions or guidance will be appreciated!

**Comment by amahendrakar** [comment]
@u1s1QS,
In order to expedite the trouble-shooting process, please provide the following details

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):
- TensorFlow version:
- Python version:
- Installed using virtualenv? pip? conda?:
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

and also a minimal code snippet, so that we can reproduce the issue on our end. Thanks!
